### ISSUE TITLE
Pkg.build fails on 0.7-master 

### ISSUE THREAD

**Issue by tbeason** [nalimilan/StringEncodings.jl#22]
I'm having trouble getting the package to work on the latest master branch. I had WinRPM troubles, those just got resolved, so WinRPM should be working correctly.

```julia
Pkg.build("StringEncodings")

# a million deprecation warnings...
# ...
# ...

INFO: Packages to install: win_iconv-dll
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near C:\Users\tbeason\.julia\v0.7\StringEncodings\deps\build.jl:976
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near C:\Users\tbeason\.julia\v0.7\StringEncodings\deps\build.jl:976
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near C:\Users\tbeason\.julia\v0.7\StringEncodings\deps\build.jl:976
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near C:\Users\tbeason\.julia\v0.7\StringEncodings\deps\build.jl:976
┌ Warning: `info()` is deprecated, use `@info` instead.
│   caller = do_install(::WinRPM.Package) at WinRPM.jl:454
└ @ WinRPM WinRPM.jl:454
INFO: Downloading: win_iconv-dll
┌ Warning: `info()` is deprecated, use `@info` instead.
│   caller = do_install(::WinRPM.Package) at WinRPM.jl:465
└ @ WinRPM WinRPM.jl:465
INFO: Extracting: win_iconv-dll
┌ Warning: `open(cmd)` now returns only a Process<:IO object.
│   caller = next at deprecated.jl:200 [inlined]
└ @ Core deprecated.jl:200

ERROR: The system cannot find the file specified.
C:\Users\tbeason\.julia\v0.7\WinRPM\cache\2\noarch%2Fmingw64-win_iconv-dll-0.0.8-3.15.noarch.cpio



System ERROR:
The system cannot find the file specified.

7-Zip [64] 16.04 : Copyright (c) 1999-2016 Igor Pavlov : 2016-10-04

Scanning the drive for archives:

┌ Error: ------------------------------------------------------------
│ # Build failed for StringEncodings
│   exception =
│    LoadError: failed process: Process(`'C:\Users\tbeason\AppData\Local\Julia-0.7.0-DEV\bin\7z.exe' x -y 'C:\Users\tbeason\.julia\v0.7\WinRPM\cache\2\noarch%2Fmingw64-win_iconv-dll-0.0.8-3.15.noarch.cpio' '-oC:\Users\tbeason\.julia\v0.7\WinRPM\deps'`, ProcessExited(2)) [2]
│    Stacktrace:
│     [1] error(::String, ::Base.Process, ::String, ::Int64, ::String) at .\error.jl:42
│     [2] pipeline_error(::Base.Process) at .\process.jl:698
│     [3] do_install(::WinRPM.Package) at C:\Users\tbeason\.julia\v0.7\WinRPM\src\WinRPM.jl:483
│     [4] do_install at C:\Users\tbeason\.julia\v0.7\WinRPM\src\WinRPM.jl:445 [inlined]
│     [5] #install#21(::Bool, ::Function, ::WinRPM.Package) at C:\Users\tbeason\.julia\v0.7\WinRPM\src\WinRPM.jl:392
│     [6] #install at .\<missing>:0 [inlined]
│     [7] #install#19 at C:\Users\tbeason\.julia\v0.7\WinRPM\src\WinRPM.jl:361 [inlined]
│     [8] #install at .\<missing>:0 [inlined] (repeats 2 times)
│     [9] (::getfield(WinRPM, Symbol("##36#37")){WinRPM.RPM})() at C:\Users\tbeason\.julia\v0.7\WinRPM\src\winrpm_bindeps.jl:42
│     [10] run(::getfield(WinRPM, Symbol("##36#37")){WinRPM.RPM}) at C:\Users\tbeason\.julia\v0.7\BinDeps\src\BinDeps.jl:484
│     [11] run(::BinDeps.SynchronousStepCollection) at C:\Users\tbeason\.julia\v0.7\BinDeps\src\BinDeps.jl:527
│     [12] satisfy!(::BinDeps.LibraryDependency, ::Array{DataType,1}) at C:\Users\tbeason\.julia\v0.7\BinDeps\src\dependencies.jl:943
│     [13] satisfy!(::BinDeps.LibraryDependency) at C:\Users\tbeason\.julia\v0.7\BinDeps\src\dependencies.jl:921
│     [14] top-level scope at C:\Users\tbeason\.julia\v0.7\BinDeps\src\dependencies.jl:976
│     [15] include(::Module, ::String) at .\boot.jl:292
│     [16] include_relative(::Module, ::String) at .\loading.jl:1012
│     [17] include at .\sysimg.jl:26 [inlined]
│     [18] include(::String) at .\loading.jl:1046
│     [19] top-level scope
│     [20] eval at .\boot.jl:295 [inlined]
│     [21] eval at .\sysimg.jl:71 [inlined]
│     [22] evalfile(::String, ::Array{String,1}) at .\loading.jl:1041 (repeats 2 times)
│     [23] #2 at .\none:15 [inlined]
│     [24] cd(::getfield(, Symbol("##2#5")){String}, ::String) at .\file.jl:59
│     [25] (::getfield(, Symbol("##1#3")))(::IOStream) at .\none:14
│     [26] #open#318(::Base.Iterators.IndexValue{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::getfield(, Symbol("##1#3")), ::String, ::Vararg{String,N} where N) at .\iostream.jl:369
│     [27] open(::Function, ::String, ::String) at .\iostream.jl:367
│     [28] top-level scope
│     [29] eval at .\boot.jl:295 [inlined]
│     [30] eval(::Module, ::Expr) at .\sysimg.jl:71
│     [31] exec_options(::Base.JLOptions) at .\client.jl:309
│     [32] _start() at .\client.jl:447
│    in expression starting at C:\Users\tbeason\.julia\v0.7\StringEncodings\deps\build.jl:976
└ @ Main none:18
┌ Warning: ------------------------------------------------------------
│ # Build error summary
│
│ StringEncodings had build errors.
│
│  - packages with build errors remain installed in C:\Users\tbeason\.julia\v0.7
│  - build the package(s) and all dependencies with `Pkg.build("StringEncodings")`
│  - build a single package by running its `deps/build.jl` script
└ @ Pkg.Entry entry.jl:651
```

Looking at the directory it is searching:
```
λ ls -1 C:\Users\tbeason\.julia\v0.7\WinRPM\cache\2
mingw64-win_iconv-dll-0.0.8-3.15.noarch.cpio
noarch%2Fmingw64-win_iconv-dll-0.0.8-3.15.noarch.rpm
repodata%2F58a3da7b6a7a7cf1c71a355252e0d9db1aab60162e9017b2235f5fa7a118660f-primary.xml
repodata%2Frepomd.xml
```

**Comment by nalimilan** [comment]
We should probably fix all deprecations on 0.7 first, and then see whether the bug is still there.

**Comment by tbeason** [comment]
This is a `WinRPM` issue still. I've fixed it locally, will submit a PR for testing soon, not sure how correct my fix was. I'll close this issue because it doesn't appear that StringEncodings was the culprit.

And yes, would still be good to eliminate all of these deprecations. :wink:
